File: MetaTracks/AcousticFingerprintingLibrary_0.4.5/SoundFingerprint/Fingerprint.cs
namespace AcousticFingerprintingLibrary_0._4._5.SoundFingerprint
{
    public class Fingerprint
    {
        public bool[] Signature { get; set; }

        // Number of fingerprint in file
        public int SequenceNumber { get; set; }

        // Where in soundfile fingerprint is
        public double Timestamp { get; set; }
        
    }

    public class HashedFingerprint
    {
        public HashedFingerprint(byte[] subFingerprint, long[] hashBins, int sequenceNumber, double sequenceAt)
        {
            SubFingerprint = subFingerprint;
            HashBins = hashBins;
            SequenceNumber = sequenceNumber;
            Timestamp = sequenceAt;
        }

        public byte[] SubFingerprint { get; set; }

        public long[] HashBins { get; set; }

        public int SequenceNumber { get; set; }

        public double Timestamp { get; set; }
    }
}

File: MetaTracksAcoustic/AudioRecognitionLibrary/AudioRecognitionLibrary/Fingerprint.swift
//
// Created by metatracks on 24/04/16.
// Copyright (c) 2016 metatracks. All rights reserved.
//

import Foundation

public class Fingerprint {
    var Signature:[Bool] = [true]
    var SequenceNumber:Int = 0
    var Timestamp:Double = 0
    init(signature: [Bool], sequenceNo: Int, timestamp: Double){}
}



public class HashedFingerprint {
    init(hashBins: [Int64], sequenceNumber: Int, sequenceAt: Double){
        HashBins = hashBins;
        SequenceNumber = sequenceNumber
        Timestamp = sequenceAt
    }
    
    init(hashBins: [Int64], timestamp: Double){
        HashBins = hashBins
        Timestamp = timestamp
        SequenceNumber = 0
    }
    
    public var HashBins:[Int64]
    public var SequenceNumber:Int
    public var Timestamp:Double
    
}
